### PR TITLE
Update temporal memory compatibility test to use C++ TM.

### LINKED ISSUE #2980: C++/Py TM compatibility test needs to be fixed to compare against C++ implementation
C++/Py TM compatibility test (#2954) currently just runs the Py TM against itself since it was created before C++ bindings were finished. Now that they are finished, it should be updated. Additionally, `TPRegion` needs to be updated to accept the new C++ TM implementation.

Requires numenta/nupic.core#839, numenta/nupic.core#840
Related to #2949

@chetan51

### PR DESCRIPTION
**Pull request by SaganBolliger**
Fixes #2980 

CI build won't pass until these are merged:
- numenta/nupic.core#841 (merged)
- numenta/nupic.core#842 (merged)

Additionally, although I believe the compatibility test is correct, it does not pass. Added skip with reference to issue #2986.

Update: skip removed. Should pass once numenta/nupic.core#846 is merged.

### CODE REVIEW

File: src/nupic/research/TP_shim.py [hunk 1]
```diff
@@ -27,10 +27,14 @@
 import numpy
 
 from nupic.research.temporal_memory import TemporalMemory
+from nupic.bindings.algorithms import TemporalMemory as TemporalMemoryCPP
 from nupic.research.fast_temporal_memory import FastTemporalMemory
 from nupic.research.monitor_mixin.temporal_memory_monitor_mixin import (
   TemporalMemoryMonitorMixin)
 
+# import inspect
+# print inspect.getargspec(TemporalMemoryCPP.__init__)
```
Comment (chetan51) [inline src/nupic/research/TP_shim.py]: Remove commented-out code

Comment (SaganBolliger) [inline src/nupic/research/TP_shim.py]: Thanks. Fixed.

File: src/nupic/research/temporal_memory.py [hunk 2]
```diff
@@ -709,6 +709,15 @@ def getMatchingCells(self):
     return self.getCellIndices(self.matchingCells)
 
 
+  def getCellsPerColumn(self):
```
Comment (chetan51) [inline src/nupic/research/temporal_memory.py]: Do we need one in bindings too?

File: tests/integration/nupic/engine/temporal_memory_compatibility_test.py [hunk 3]
```diff
@@ -39,10 +39,8 @@ def testTMPyCpp(self):
     results1 = createAndRunNetwork(TPRegion,
                                    "bottomUpOut",
                                    checkpointMidway=False,
-                                   temporalImp="tm_py")
+                                   temporalImp="tm_cpp")
 
-    # temporalImp="tm_py" here is a temporary placeholder value until C++ TM is
```
Comment (chetan51) [inline tests/integration/nupic/engine/temporal_memory_compatibility_test.py]: No, we want to compare `tm_py` and `tm_cpp` in this test.

Comment (SaganBolliger) [inline tests/integration/nupic/engine/temporal_memory_compatibility_test.py]: The old comment was incorrect. The test does compare `tm_py` and `tm_cpp`

File: src/nupic/research/temporal_memory.py (hunk 2 above)
Comment (SaganBolliger) [inline src/nupic/research/temporal_memory.py]: This was already in the C++. I added in the Python for API compatibility.

File: tests/integration/nupic/engine/temporal_memory_compatibility_test.py (hunk 3 above)
Comment (chetan51) [inline tests/integration/nupic/engine/temporal_memory_compatibility_test.py]: Right, so we should remove this comment then

File: src/nupic/research/temporal_memory.py (hunk 2 above)
Comment (chetan51) [inline src/nupic/research/temporal_memory.py]: Got it.

File: tests/integration/nupic/engine/temporal_memory_compatibility_test.py (hunk 3 above)
Comment (SaganBolliger) [inline tests/integration/nupic/engine/temporal_memory_compatibility_test.py]: It is being removed by this commit.

Comment (chetan51) [inline tests/integration/nupic/engine/temporal_memory_compatibility_test.py]: Oh my bad, I misread that.

**Comment by chetan51** [comment]
:+1:

**Comment by chetan51** [comment]
Fixed by https://github.com/numenta/nupic.core/pull/846.

**Comment by SaganBolliger** [comment]
Removed the skip for the test now that @chetan51 has fixed the TM compatibility issues.

**Comment by chetan51** [comment]
@SaganBolliger It will still fail though since https://github.com/numenta/nupic.core/pull/846 hasn't been merged yet.

**Comment by scottpurdy** [comment]
Replaced by #3005

**Pull request closed without merging**